Update tool IceCube to 0.0.1+galaxy0

dsavchenko opened the pull request.

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/0b8472472245415fa2d4704b1a45a7f67514495c

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/2203796d07ed8f7ea07d99377c6c72566930d684

volodymyrss commented on tools/icecube/icecube_astro_tool.xml:
```diff
@@ -0,0 +1,108 @@
+    <requirement type="package" version="0.70.16">multiprocess</requirement>
+    <requirement type="package" version="1.2.12">oda-api</requirement>
+    <requirement type="package" version="7.16.1">nbconvert</requirement>
+    <!--Requirements string 'skyllh
```
why can it not be, @dsavchenko ?

dsavchenko replied:
Seems it's not in conda
So currently no way to create the tool this way...

volodymyrss replied:
As a workaround, can you just do pip install while running the tool?..

dsavchenko replied:
It will work in our staging installation, I think. Might not be fully universal with other job runners.

But this is a really weird way, not sure, we want to include it in the generator

volodymyrss replied:
On the other hand, I am not sure why galaxy does not allow pip. renkulab for example [allows this](https://github.com/SwissDataScienceCenter/renku-project-template/blob/ee4cf00c52c6cd45363a40b55540343e9ecf70af/python-minimal/Dockerfile#L45-L50) and it works fine for a large number of projects.
@bgruening , I am sure you saw this question before, and maybe even we asked, but is it really not possible to use pip packages in galaxy?

Anyway, since we auto-generate tool here, we can consider this extra pip install as part of bootstrapping, the user never sees it directly.

> It will work in our staging installation, I think. Might not be fully universal with other job runners.

Why not? Because there is no writable directory? Could be even --user assuming there is a writable home. Or it can be created.

> But this is a really weird way, not sure, we want to include it in the generator

It could be a fallback, allowing to show what capacity is provided by adding the tool to galaxy. This can motivate the package authors to put to conda. We probably do not want to push it to toolsched. Maybe the PR can be labeled as so. PRs with incomplete dependencies should be distingiushed.

bgruening replied:
Sorry, only saw this no. 
dependency managers are just another plugin layer in Galaxy. brew, conda, modules ... ofc you can add pip if you like. The question we need to ask is where to stop, do we now need to support all language-specific package managers? How good are they? We still do want to move to containers, isn't it? How do we ensure containers are available for all the packages?

On the other side, adding pypi packages to conda-forge is so trivial nowadays and the entire OS community benefits from it.

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/7c79518819ce2ce283940433aa1bafcf8d193bd0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/2a371f6721b313c6f0d9b02a5acda4a558bf381f

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/f687835f21d7513b61f597c21fcf04013669c95e

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/09b7d2063c9cbef533640e69c93d7a320627325e

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/74c7f95cd48ef2981a5f4482a0590e21d476cec2

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/0645d90e4c2bb792ea2fa61a10afa0f377073f5c

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/218815da526419de5039398d69b2eb291d29d1fe

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/1dcab70fee2d6928f0665ed0f9b54472b6226d5a

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/283a383a6eb94c1b82ab35f2a9ff42ec6b1375d3

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/28a8eb6d1719798a2b36ffe1e1833353490418a0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/1a8ed7f69e38e856f809db095ab66793683158aa

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/aa5e8eb3ecd1f0fee662283ea2de5c20bbadbfb5

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/8f7e9554cca73c05fe50aba48c5e17e7cacd70c0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/a29333ddf2c2a8fe4a4c61b97e24fd1d040cf492

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/4d871a71b258053b6495d310d819b17ecad16b5c

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/3e1a0ab3f6a88070f10f75f623d268c9a94037b5

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/83bcf3b42fdea89d3527c77ec59b727f8a0d3da3

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/8dcaf61f2d00866b4dde184d5a29760b5b1f6384

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/46bcf8b9a70a01af8566799e726582c0e94be901

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/e64daae0050c49445706f177ba01f232426c00f0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/57096d2ae3d48ec826e65e483f9009cdf3790ecd

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/c280ff2275dba5acc628f1f5520bd2145eb0d831

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/cae9040f576047780b755b878b5d9e7fa1cd5354

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/86901b5c4414b79d9bbc3067138afdc27d79450b

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/9217abe9663addfba6d0ea4914257ee3df4c40e4

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/c24b16fa2174ebe2f3f2de37dc8766ce534656bf

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/d477302c0ecd42149058509078d9a734c262625b

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/72e471bc1b2257100476dc68def81e7fb88d3c61

dsavchenko committed:
Merge branch 'main' into auto-update-galaxy-tool-icecube-v0.0.1-galaxy0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/1fab063aed5b1c4054423f1e2fc0f326cc0e3e5b

dsavchenko committed:
Merge branch 'main' into auto-update-galaxy-tool-icecube-v0.0.1-galaxy0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/d8d8536cf738e851cc4e8e864ed6b6b9edfb6951

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/d8d8536cf738e851cc4e8e864ed6b6b9edfb6951

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/d8d8536cf738e851cc4e8e864ed6b6b9edfb6951

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/a166bf242b8ff38071372c185b4200187545fdf2

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/24af5ed3ec3e69e4aab32aa104990256114c191b

dsavchenko committed:
Merge branch 'main' into auto-update-galaxy-tool-icecube-v0.0.1-galaxy0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/21fb4d0b7bbb595c21eee0f21afab6d99c763e2d

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/e99209d0e2cc155f516c890f0e597807d736f0d5

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/5fb6627c27e3cea61a703748f0724f06ef9ca4f0

dsavchenko committed:
Merge branch 'main' into auto-update-galaxy-tool-icecube-v0.0.1-galaxy0

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/8c6fb57ff6fecf5f52de33e0b18baa62c591ece2

Commit:
automatic update

following https://gitlab.renkulab.io/astronomy/mmoda/icecube/-/commit/e9a0c6ab65ab9dc458e7824e34bbd8b8563382d6
